python/ray/rllib/agents/dqn/dqn.py

```diff
@@ -206,6 +229,12 @@ def _train(self):
                 "max_exploration": max(exp_vals),
                 "num_target_updates": self.num_target_updates,
             }, **self.optimizer.stats()))
+
+        if self.config["evaluation_interval"]:
+            if self.iteration % self.config["evaluation_interval"] == 0:
+                self.evaluation_metrics = self._evaluate()
+            result.update(self.evaluation_metrics)
```

Comment:
Does this work in Tune? I think this might affect the CSV logger

Reply:
(not sure but it's because the CSV logger is not robust to changing field sizes and probably should be fixed)